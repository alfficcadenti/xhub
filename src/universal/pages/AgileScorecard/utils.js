import moment from 'moment';

export const labelFormat = ({count, total, label}) => count ? `${label}  (${count} of ${total} selected)` : `Select ${label}...`;

export const formatPieData = (data) => Array.isArray(data) && data.map((x) => ({name: x.type_of_work, value: x.ticket_count})) || [];

export const formatLineChartData = (data) => Array.isArray(data) && data
    .map((x) => ({name: moment(x.date).format('YYYY-MM-DD'), 'open bugs': x.open_bugs_count || 0, 'closed bugs': x.closed_bugs_count || 0}))
    .sort((a, b) => (a.name).localeCompare(b.name)) || [];

export const formatTooltipData = (data) => Array.isArray(data) && data
    .reduce((a, x) => ({...a, [moment(x.date).format('YYYY-MM-DD')]: {'open bugs': x.open_bugs_ticket_ids?.sort(), 'closed bugs': x.closed_bugs_ticket_ids?.sort()}}), {}) || {};

export const listOfBugs = (dataObj, type) => dataObj?.[type] || [];
