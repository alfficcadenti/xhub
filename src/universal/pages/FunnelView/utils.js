import {OPXHUB_SUPPORT_CHANNEL} from '../../constants';


export const getErrorMessage = (selectedBrand) => {
    return `Page views for ${selectedBrand} is not yet available.
        The following brands are supported at this time: "Expedia", "Hotels.com Retail", and "Vrbo Retail".
        If you have any questions, please ping ${OPXHUB_SUPPORT_CHANNEL} or leave a comment via our Feedback form.`;
};
