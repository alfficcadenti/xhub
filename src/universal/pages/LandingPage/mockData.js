// export const CSRData = [
//     {brandName: 'BEX', CSRTrend: [1, 2, 3, 6, 7, 3, 4, 10], CSRByDevice: {mobile: 90, tablet: 99, desktop: 60}},
//     {brandName: 'Hotels.com', CSRTrend: [1, 2, 3, 6, 5, 11, 10, 6], CSRByDevice: {mobile: 10, tablet: 99, desktop: 60}},
//     {brandName: 'Vrbo', CSRTrend: [10, 9, 3, 4, 5, 6, 4, 9], CSRByDevice: {mobile: 97, tablet: 50, desktop: 60}},
// ];

export const CSRData = [
    {
        'time': '2020-05-21T10:30:00.000Z', BEX: 0.99, Vrbo: 0.1, 'Hotels.com': 0.1,
    },
    {
        'time': '2020-05-21T11:30:00.000Z', BEX: 0.9, Vrbo: 0.1, 'Hotels.com': 0.2,
    },
    {
        'time': '2020-05-21T12:30:00.000Z', BEX: 0.9, Vrbo: 0.1, 'Hotels.com': 0.3,
    },
    {
        'time': '2020-05-21T13:30:00.000Z', BEX: 0.9, Vrbo: 1, 'Hotels.com': 0.7,
    },
    {
        'time': '2020-05-21T14:30:00.000Z', BEX: 0.9, Vrbo: 0.9, 'Hotels.com': 0.98,
    },
    {
        'time': '2020-05-21T15:30:00.000Z', BEX: 0.5, Vrbo: 0.7, 'Hotels.com': 0.8,
    },
    {
        'time': '2020-05-21T16:30:00.000Z', BEX: 0.99, Vrbo: 0.89, 'Hotels.com': 0.72,
    },
];

export const bookingsData = [
    {
        'time': '2020-05-21T10:30:00.000Z', BEX: 4000, Vrbo: 2400, 'Hotels.com': 2400,
    },
    {
        'time': '2020-05-21T10:30:00.000Z', BEX: 3000, Vrbo: 1398, 'Hotels.com': 2210,
    },
    {
        'time': '2020-05-21T10:30:00.000Z', BEX: 2000, Vrbo: 9800, 'Hotels.com': 2290,
    },
    {
        'time': '2020-05-21T10:30:00.000Z', BEX: 2780, Vrbo: 3908, 'Hotels.com': 2000,
    },
    {
        'time': '2020-05-21T10:30:00.000Z', BEX: 1890, Vrbo: 4800, 'Hotels.com': 2181,
    },
    {
        'time': '2020-05-21T10:30:00.000Z', BEX: 2390, Vrbo: 3800, 'Hotels.com': 2500,
    },
    {
        'time': '2020-05-21T10:30:00.000Z', BEX: 3490, Vrbo: 4300, 'Hotels.com': 2100,
    },
];