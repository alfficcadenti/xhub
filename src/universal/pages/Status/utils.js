export const check = (array1, array2) => {
    if (array1.length === array2.length) {
        // eslint-disable-next-line no-unused-expressions
        return array1.every(((result) => {
            return array2.includes(result);
        }));
    }
    return false;
};