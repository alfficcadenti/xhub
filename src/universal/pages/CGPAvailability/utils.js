import moment from 'moment';
import {getOrDefault} from '../../utils';
import {getValue} from '../utils';
import {THRESHOLDS, DATE_FORMAT} from './constants.js';
import AvailabilityCell from './AvailabilityCell';
import React from 'react';

// eslint-disable-next-line complexity
export const defineClassByValue = (value) => {
    if (typeof value !== 'number') {
        return '';
    } else if (value >= THRESHOLDS.high) {
        return 'positive';
    } else if (value >= THRESHOLDS.medium) {
        return 'attention';
    } else if (value < THRESHOLDS.medium) {
        return 'negative';
    }
    return '';
};

export const formattedValue = (value) => Number.isFinite(value) ? Math.floor(value * 100) / 100 : '-';

export const periodAvailability = (availabilities) => {
    const periodSummary = availabilities?.reduce((acc, curr) => {
        acc.totalRequests += (curr.requestCount || 0);
        acc.errorsCount += (curr.errorCount || 0);
        return acc;
    }, {totalRequests: 0, errorsCount: 0});
    return formattedValue((periodSummary.totalRequests - periodSummary.errorsCount) / periodSummary.totalRequests * 100);
};

export const extractColumns = (data = [], dateTimeFormat) => ['Application', ...data[0]?.availabilities?.map((item) => item && moment(item.timestamp).format(dateTimeFormat)) || [], 'Availability'];

export const exactAvailability = (availability, requestCount) => {
    if (requestCount) {
        return availability;
    }
    return null;
};

export const mapAvailabilityRow = (row = {}, handleClick, dateTimeFormat = DATE_FORMAT) => {
    const app = getOrDefault(row, 'applicationName');
    const res = Array.isArray(row?.availabilities) &&
        Object.assign(
            {Application: app},
            {Availability: <AvailabilityCell value={periodAvailability(row?.availabilities)} applicationName={app} handleClick={handleClick}/>},
            {avgValue: periodAvailability(row?.availabilities)},
            ...row?.availabilities.map((x) => ({[moment(x.timestamp).format(dateTimeFormat)]: <AvailabilityCell value={exactAvailability(x.availability, x.requestCount)} applicationName={app} handleClick={handleClick}/>}))) || {};
    return res;
};

export const getAppErrorsDataForChart = (applicationName = '', availability = []) => Array.isArray(availability) && availability?.find((x) => x.applicationName === applicationName)?.availabilities?.map((x) => ({name: moment(x?.timestamp).format(DATE_FORMAT), '5xx Errors': x?.errorCount})) || [];

export const getSelectedRegions = (regionsObj) => (regionsObj?.length ? regionsObj.filter((x) => x.counted && x.checked).map((x) => x.name) : '');

export const getPresets = () => [
    {text: 'Last 24 Hours', value: getValue(23.98, 'hours')},
    {text: 'Last 3 days', value: getValue(3, 'days')},
    {text: 'Last 7 days', value: getValue(7, 'days')},
    {text: 'Last 15 days', value: getValue(15, 'days')},
    {text: 'Last 30 days', value: getValue(30, 'days')},
];
