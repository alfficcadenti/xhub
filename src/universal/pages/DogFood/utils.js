import qs from 'query-string';
import {validDateRange, getUrlParam, getValue} from '../utils';
import moment from 'moment';
import {getOrDefault} from '../../utils';
import {DATE_FORMAT, ALL_PROJECTS_OPTION, ALL_STATUSES_OPTION} from '../../constants';


// eslint-disable-next-line complexity
export const getQueryValues = (search) => {
    const {
        start,
        end,
        status,
        project
    } = qs.parse(search);
    const isValidDateRange = validDateRange(start, end);
    return {
        initialStart: isValidDateRange
            ? moment(start).format(DATE_FORMAT)
            : moment().subtract(1, 'month').startOf('minute').format(DATE_FORMAT),
        initialEnd: isValidDateRange
            ? moment(end).format(DATE_FORMAT)
            : moment().format(DATE_FORMAT),
        initialTimeRange: isValidDateRange ? 'Custom' : 'Last 7 days',
        initialStatus: status || ALL_STATUSES_OPTION,
        initialProject: project || ALL_PROJECTS_OPTION,
    };
};

export const getActiveIndex = (pathname = '') => {
    if (pathname.includes('dog-food/overview')) {
        return 0;
    }
    if (pathname.includes('dog-food/issues')) {
        return 1;
    }
    return 0;
};

export const mapActiveIndexToTabName = (idx) => {
    if (idx === 1) {
        return 'issues';
    }
    return 'overview';
};

export const generateUrl = (
    activeIndex,
    selectedBrands,
    startDate,
    endDate,
    selectedStatus,
    selectedProject
) => {
    return `/dog-food/${mapActiveIndexToTabName(activeIndex)}`
        + `?selectedBrand=${selectedBrands[0]}`
        + `&start=${startDate}`
        + `&end=${endDate}`
        + `${getUrlParam('project', selectedProject, ALL_PROJECTS_OPTION)}`
        + `${getUrlParam('status', selectedStatus, ALL_STATUSES_OPTION)}`;
};

const formatDate = (dateString) => dateString && moment(dateString).isValid ? moment(dateString).format('YYYY-MM-DD HH:mm') : '-';

export const mapIssues = (x) => {
    return ({
        'Ticket Id': `<a href="${x.url}" target="_blank">${x.id}</a>`,
        Priority: getOrDefault(x, 'priority'),
        Severity: getOrDefault(x, 'severity'),
        Brand: getOrDefault(x, 'brand'),
        Summary: getOrDefault(x, 'summary'),
        Project: getOrDefault(x, 'project'),
        'Project Key': getOrDefault(x, 'project_key'),
        Status: getOrDefault(x, 'status'),
        Tag: getOrDefault(x, 'tag'),
        Resolution: getOrDefault(x, 'resolution'),
        'Impacted Brand': getOrDefault(x, 'impacted_brand'),
        'Last Update': formatDate(x.updated_date_time),
        'Open Date': formatDate(x.open_date),
        'Resolved Date': formatDate(x.resolved_date),
        'Labels': x.labels && Array.isArray(x.labels) ? x.labels.slice(',') : []
    });
};

export const getPresets = () => [
    {text: 'Last 24 hours', value: getValue(24, 'hours')},
    {text: 'Last 7 days', value: getValue(7, 'days')},
    {text: 'Last 14 days', value: getValue(14, 'days')},
    {text: 'Last 30 days', value: getValue(30, 'days')},
    {text: 'Last 60 days', value: getValue(60, 'days')},
    {text: 'Last 90 days', value: getValue(90, 'days')},
    {text: 'Last 1 year', value: getValue(1, 'year')}
];