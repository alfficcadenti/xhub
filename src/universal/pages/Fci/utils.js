import React from 'react';
import moment from 'moment';
import qs from 'query-string';
import {SVGIcon} from '@homeaway/react-svg';
import {EDIT__16} from '@homeaway/svg-defs';
import DataTable from '../../components/DataTable';
import {LOB_LIST} from '../../constants';
import {TRACE_TABLE_COLUMNS, SITES, CATEGORY_OPTION} from './constants';
import {EXPEDIA_PARTNER_SERVICES_BRAND, EXPEDIA_BRAND, OPXHUB_SUPPORT_CHANNEL} from '../../constants';
import {getOrDefault} from '../../utils';
import {validDateRange} from '../utils';


export const getInitialSelectData = (initialOption) => ({
    start: null,
    end: null,
    options: [{label: initialOption, value: initialOption}]
});

export const getBrandSites = (brand) => SITES[brand] || ['travel.chase.com'];

export const getIsSupportedBrand = (selectedBrands) => [EXPEDIA_PARTNER_SERVICES_BRAND, EXPEDIA_BRAND].includes(selectedBrands[0]);

export const getUnsupportedBrandMsg = (selectedBrands) => `FCIs for ${selectedBrands[0]} is not yet available. `
    + `For now only ${EXPEDIA_PARTNER_SERVICES_BRAND} and ${EXPEDIA_BRAND} is supported. `
    + `If you have any questions, please ping ${OPXHUB_SUPPORT_CHANNEL} or leave a comment via our Feedback form.`;

// eslint-disable-next-line complexity
export const shouldFetchData = (prev, start, end, selectedSite, selectedLob, chartProperty, selectedErrorCode, hideIntentionalCheck) => (
    !prev.start
    || !prev.end
    || start.isBefore(prev.start)
    || end.isAfter(prev.end)
    || prev.selectedSite !== selectedSite
    || prev.selectedLob !== selectedLob
    || prev.chartProperty !== chartProperty
    || prev.selectedErrorCode !== selectedErrorCode
    || prev.hideIntentionalCheck !== hideIntentionalCheck
);

// eslint-disable-next-line complexity
export const getQueryValues = (search, brand = 'Expedia') => {
    const {tab, from, to, lobs, code, sites, hide_intentional: hideIntentional, search_id: searchId, bucket, id, deltaUsersId} = qs.parse(search);
    const isValidDateRange = validDateRange(from, to);
    return {
        initialStart: isValidDateRange ? moment(from) : moment().subtract(24, 'hours').startOf('minute'),
        initialEnd: isValidDateRange ? moment(to) : moment(),
        initialTimeRange: isValidDateRange ? 'Custom' : 'Last 24 Hours',
        initialLobs: lobs
            ? lobs.split(',').map((l) => LOB_LIST.find(({value}) => value === l)).filter((l) => l)
            : [],
        initialSite: sites
            ? sites.split(',')
            : [getBrandSites(brand)[0]],
        initialErrorCode: code ? code.split(',') : [],
        initialHideIntentionalCheck: hideIntentional === 'true',
        initialSearchId: searchId || '',
        initialDeltaUsersId: deltaUsersId || '',
        initialSelectedId: id || '',
        initialIndex: ['0', '1'].includes(tab) ? Number(tab) : 0,
        initialBucket: bucket && moment(bucket).isValid ? bucket : null
    };
};

export const stringifyQueryParams = (selectedValue) => Array.isArray(selectedValue) ? selectedValue.map((v) => v.value || v).join(',') : selectedValue;

export const getFciQueryString = (start, end, selectedErrorCode, selectedSite, selectedLob, hideIntentionalCheck, chartProperty) => {
    const dateQuery = `from=${start.toISOString()}&to=${end.toISOString()}`;
    const errorProperty = chartProperty === CATEGORY_OPTION ? 'category' : 'code';
    const errorQuery = selectedErrorCode?.length ? `&${errorProperty}=${stringifyQueryParams(selectedErrorCode)}` : '';
    const siteQuery = selectedSite?.length ? `&sites=${stringifyQueryParams(selectedSite)}` : '';
    const lobQuery = selectedLob?.length ? `&line_of_business=${stringifyQueryParams(selectedLob)}` : '';
    const hideIntentionalCheckQuery = `&hide_intentional=${hideIntentionalCheck}`;
    return `${dateQuery}${errorQuery}${siteQuery}${lobQuery}${hideIntentionalCheckQuery}`;
};

// eslint-disable-next-line complexity
export const getHistoryQueryString = (selectedBrands, start, end, selectedErrorCode, selectedSite,
    hideIntentionalCheck, chartProperty, searchId, activeIndex, selectedBucket, id) => {
    const brandQuery = `selectedBrand=${selectedBrands[0]}`;
    const dateQuery = `&from=${start.toISOString()}&to=${end.toISOString()}`;
    const errorProperty = chartProperty === CATEGORY_OPTION ? 'code' : 'code';
    const errorQuery = selectedErrorCode && selectedErrorCode.length ? `&${errorProperty}=${stringifyQueryParams(selectedErrorCode)}` : '';
    const siteQuery = selectedSite ? `&sites=${stringifyQueryParams(selectedSite)}` : '';
    const hideIntentionalCheckQuery = `&hide_intentional=${hideIntentionalCheck}`;
    const searchQuery = searchId ? `&search_id=${searchId}` : '';
    const indexQuery = `&tab=${activeIndex}`;
    const bucketQuery = selectedBucket ? `&bucket=${selectedBucket}` : '';
    const idQuery = id ? `&id=${id}` : '';
    return `${brandQuery}${dateQuery}${errorQuery}${siteQuery}${hideIntentionalCheckQuery}`
        + `${searchQuery}${indexQuery}${bucketQuery}${idQuery}`;
};

const getTagValue = (tags, property) => {
    const foundIdx = tags.findIndex(({key}) => (key || '').includes(property));
    if (foundIdx > -1) {
        return tags[foundIdx].value || '-';
    }
    return '-';
};

export const traceHasError = (t) => {
    const tags = t.tags || [];
    const errorIdx = tags.findIndex(({key}) => key === 'error');
    const Error = errorIdx > -1 ? tags[errorIdx].value : '-';
    return Error === 'true';
};

export const getTraceCounts = (traces) => traces.reduce((acc, curr) => {
    if (curr.Error === 'true') {
        acc.errors++;
    }
    acc.total++;
    return acc;
}, {errors: 0, total: 0});

// eslint-disable-next-line complexity
export const mapTrace = (t) => {
    const tags = t.tags || [];
    const hasError = traceHasError(t);
    const result = {
        Service: getOrDefault(t, 'service_name'),
        Operation: getOrDefault(t, 'operation_name'),
        Error: String(hasError),
        'External Error Code': '-',
        'External Description': '-',
        traces: t.traces,
        Traces: !t.traces || !t.traces.length ? null : (
            <DataTable
                title={`Trace Log (ID=${t.traceId})`}
                data={t.traces.map(mapTrace)}
                rules={[{column: 'Error', setClass: (val) => val === 'true' ? 'error-cell' : 'success-cell'}]}
                columns={TRACE_TABLE_COLUMNS}
                expandableColumns={['Traces']}
            />
        )
    };
    if (hasError) {
        result['External Error Code'] = getTagValue(tags, 'externalerrorcode');
        result['External Description'] = getTagValue(tags, 'externalerrordescription');
    }
    return result;
};

export const mapComment = (row) => ({
    Created: getOrDefault(row, 'timestamp'),
    Author: getOrDefault(row, 'author'),
    Comment: getOrDefault(row, 'comment'),
    'Is FCI': String(row.isFci)
});

export const mapFci = (row = {}) => {
    const {fci = {}, category = []} = JSON.parse(JSON.stringify(row));
    return {
        Created: fci.timestamp ? moment(fci.timestamp).format('YYYY-MM-DD HH:mm') : '-',
        Session: getOrDefault(fci, 'session_id'),
        Trace: getOrDefault(fci, 'trace_id'),
        Failure: getOrDefault(fci, 'failure'),
        'Intentional': getOrDefault(fci, 'is_intentional'),
        'Error Code': getOrDefault(fci, 'error_code'),
        Site: getOrDefault(fci, 'site'),
        LOB: getOrDefault(fci, 'lob'),
        TPID: getOrDefault(fci, 'tp_id'),
        EAPID: getOrDefault(fci, 'eap_id'),
        'SiteID': getOrDefault(fci, 'site_id'),
        Category: category.join(', ') || '-',
        LoB: (LOB_LIST.find((l) => l.value === fci.line_of_business) || {label: '-'}).label,
        'Device User Agent ID': getOrDefault(fci, 'dua_id'),
        Comment: getOrDefault(fci, 'comment'),
        'Is FCI': String(fci.is_fci),
        recordedSessionUrl: getOrDefault(row, 'recorded_session_url'),
        traces: (fci.traces || []).map(mapTrace)
    };
};

export const mapDeltaUser = (row = {}) => {
    return {
        Created: row.timestamp ? moment(row.timestamp).format('YYYY-MM-DD HH:mm') : '-',
        Brand: getOrDefault(row, 'brand'),
        LoB: (LOB_LIST.find((l) => l.value === row.lineOfBusiness) || {label: '-'}).label,
        'Funnel Step': getOrDefault(row, 'metricName'),
    };
};

export const getTableData = (data, onOpenEdit) => {
    const result = data
        .map((row) => {
            const fci = mapFci(row);
            const editClickHandler = () => onOpenEdit(fci);
            const traceCounts = getTraceCounts(fci.traces);
            fci.Traces = (
                <div
                    className="modal-link"
                    role="button"
                    tabIndex={0}
                    onClick={editClickHandler}
                    onKeyUp={editClickHandler}
                >
                    {`Open Log (${traceCounts.errors} error${traceCounts.errors === 1 ? '' : 's'})`}
                </div>
            );
            fci.Edit = (
                <div
                    className="modal-link"
                    role="button"
                    tabIndex={0}
                    onClick={editClickHandler}
                    onKeyUp={editClickHandler}
                >
                    <SVGIcon usefill markup={EDIT__16} />
                </div>
            );
            return fci;
        });
    result.sort((a, b) => b.Created.localeCompare(a.Created));
    return result;
};

export const getFilteredTraceData = (data) => (
    (data || []).filter(({Error}) => Error === 'true')
);

export const getDeltaUserTableData = (data) => {
    if (data && data?.length) {
        const result = data
            .map((row) => mapDeltaUser(row));
        result.sort((a, b) => b.Created.localeCompare(a.Created));
        return result;
    }
    return [];
};