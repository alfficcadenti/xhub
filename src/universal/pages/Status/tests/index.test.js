import React from 'react';
import {render, act} from '@testing-library/react';
import {shallow} from 'enzyme';
import StatusPage from '../index';
import {compareObjArraysElements} from '../utils';
import {INCIDENTS_EXPECTED_DATA} from './mockData/mockData';

describe('Status Page', () => {
    let wrapper;
    const c = new AbortController();

    beforeEach(() => {
        fetch.resetMocks();
    });

    afterEach(() => {
        wrapper.unmount();
    });

    it('renders successfully', () => {
        wrapper = shallow(<StatusPage />);
        expect(wrapper).toHaveLength(1);
    });

    it('After 5 seconds waiting for a response, the request should timeout and show an alert icon', async () => {
        jest.useFakeTimers();

        fetch.mockResponseOnce(async () => {
            jest.advanceTimersByTime(6000);
        });
        setTimeout(() => c.abort(), 5000);

        await act(async () => {
            wrapper = render(<StatusPage />);
        });

        await expect(() => fetch('/v1/checkout-failures/sites?from=2021-10-26T16:53:00Z&to=2021-10-26T16:53:06Z', {signal: c.signal})).toThrow('The operation was aborted.');

        expect(wrapper).toMatchSnapshot();
        jest.useRealTimers();
    });

    it('renders fail icon when incorrect response data doesn\'t match', async () => {
        fetch.mockImplementation(() => {
            return Promise.resolve({
                ok: true,
                json: () => Promise.resolve('www.orbitz.com')
            });
        });

        await act(async () => {
            wrapper = render(<StatusPage />);
        });
        expect(wrapper).toMatchSnapshot();
    });

    it('renders success icon when correct response data matches', async () => {
        fetch.mockImplementation(() => {
            return Promise.resolve({
                ok: true,
                json: () => Promise.resolve(INCIDENTS_EXPECTED_DATA)
            });
        });

        await act(async () => {
            wrapper = render(<StatusPage />);
        });
        expect(wrapper).toMatchSnapshot();
    });

    it('fetches and calls all services', async () => {
        fetch.mockImplementation(() => {
            return Promise.resolve({
                ok: true,
                json: () => Promise.resolve()
            });
        });

        await act(async () => {
            wrapper = render(<StatusPage />);
        });

        expect(fetch.mock.calls).toEqual([
            ['/v1/checkout-failures/sites?from=2021-10-26T16:53:00Z&to=2021-10-26T16:53:06Z', {'signal': c.signal}],
            ['v1/incidents?from_datetime=2021-10-28T05:31:00Z&to_datetime=2021-10-28T16:31:00Z', {'signal': c.signal}]
        ]);
    });

    it('compareObjArraysElements returns true when two arrays match', async () => {
        const arrayOne = ['www.expedia.com', 'www.travelocity.com'];
        const arrayTwo = ['www.travelocity.com', 'www.expedia.com'];

        const verifyCheck = compareObjArraysElements(arrayOne, arrayTwo);
        expect(verifyCheck).toEqual(true);
    });

    it('compareObjArraysElements returns false when two arrays doesn\'t match', async () => {
        const arrayOne = ['www.expedia.com', 'www.travelocity.com'];
        const arrayTwo = ['www.orbitz.com', 'www.ebookers.com'];

        const verifyCheck = compareObjArraysElements(arrayOne, arrayTwo);
        expect(verifyCheck).toEqual(false);
    });

    it('compareObjArraysElements returns false when 2 objects doesn\'t match', async () => {
        const objectOne = {
            'booking_impact': 'Moderate Booking Impact',
            'brand': 'Expedia Services',
            'degradation_outage': 'Degradation',
            'duration': '106',
            'end_date': '2021-10-18 19:11:00.0000000 +00:00',
            'environment': 'Production',
            'estimated_gross_loss': null,
            'estimated_order_loss': null,
            'estimated_revenue_loss': null
        };
        const objectTwo = {
            'booking_impact': 'Moderate Booking Impact',
            'brand': 'Total Retail',
            'degradation_outage': 'Degradation',
            'duration': '106',
            'end_date': '2021-10-18 19:11:00.0000000 +00:00',
            'environment': 'Production',
            'estimated_gross_loss': null,
            'estimated_order_loss': null,
            'estimated_revenue_loss': null
        };
        const arr1 = [objectOne];
        const arr2 = [objectTwo];
        const verifyCheck = compareObjArraysElements(arr1, arr2);
        expect(verifyCheck).toEqual(false);
    });

    it('compareObjArraysElements returns true when 2 objects match', async () => {
        const objectOne = {
            'booking_impact': 'Moderate Booking Impact',
            'brand': 'Total Retail',
            'degradation_outage': 'Degradation',
            'duration': '106',
            'end_date': '2021-10-18 19:11:00.0000000 +00:00',
            'environment': 'Production',
            'estimated_gross_loss': null,
            'estimated_order_loss': null,
            'estimated_revenue_loss': null
        };
        const objectTwo = {
            'booking_impact': 'Moderate Booking Impact',
            'brand': 'Total Retail',
            'degradation_outage': 'Degradation',
            'duration': '106',
            'end_date': '2021-10-18 19:11:00.0000000 +00:00',
            'environment': 'Production',
            'estimated_gross_loss': null,
            'estimated_order_loss': null,
            'estimated_revenue_loss': null
        };
        const arr1 = [objectOne];
        const arr2 = [objectTwo];
        const verifyCheck = compareObjArraysElements(arr1, arr2);
        expect(verifyCheck).toEqual(true);
    });
});
