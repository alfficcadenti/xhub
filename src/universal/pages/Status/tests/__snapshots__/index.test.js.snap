// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status Page matches the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="status-page-container"
      >
        <h1
          class="page-title"
          data-testid="title"
        >
          Status Page
        </h1>
        <div
          class="Datatable Datatable--density-default"
        >
          <table
            aria-rowcount="1"
            class="Datatable__table Datatable__table--col-fixed"
          >
            <colgroup>
              <col
                style="width: 33.333333333333336%;"
              />
              <col
                style="width: 50%;"
              />
              <col
                style="width: 16.666666666666668%;"
              />
            </colgroup>
            <thead
              class="Datatable__header"
            >
              <tr
                role="row"
              >
                <th
                  aria-sort="none"
                  class="header__cell Datatable__cell"
                  role="columnheader"
                  scope="col"
                >
                  <div
                    class="header__name"
                  >
                    Service Name
                  </div>
                </th>
                <th
                  aria-sort="none"
                  class="header__cell Datatable__cell"
                  role="columnheader"
                  scope="col"
                >
                  <div
                    class="header__name"
                  >
                    Endpoint
                  </div>
                </th>
                <th
                  aria-sort="none"
                  class="header__cell Datatable__cell"
                  role="columnheader"
                  scope="col"
                >
                  <div
                    class="header__name"
                  >
                    Status
                  </div>
                </th>
              </tr>
            </thead>
            <tbody
              class="Datatable__body"
            >
              <tr
                class="Datatable__row"
              >
                <td
                  class="row__cell Datatable__cell"
                >
                  <a
                    href="https://github.expedia.biz/eg-reo-opex/eg-fci-data-service"
                    target="_blank"
                  >
                    eg-fci-data-service
                  </a>
                  <img
                    alt="github"
                    height="24"
                    title="github"
                    width="24"
                  />
                </td>
                <td
                  class="row__cell Datatable__cell"
                >
                  /v1/checkout-failures/sites
                </td>
                <td
                  class="row__cell Datatable__cell"
                >
                  <div
                    data-testid="successIcon"
                  >
                    <span
                      class="SVGIcon success-icon"
                    >
                      <svg
                        data-id="SVG_SUCCESS__24"
                        focusable="false"
                        height="24"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <g
                          fill="none"
                          stroke-linecap="round"
                          stroke-linejoin="round"
                        >
                          <path
                            d="M2.8 12a9.3 9.3 0 0 1 12.7-8.5 9.3 9.3 0 0 1 5 12 9.3 9.3 0 0 1-12 5A9.3 9.3 0 0 1 2.8 12z"
                          />
                          <path
                            d="M13.2 12.5l-2.9 2.8-2.8-2.8M16.5 9.1l-3.3 3.4"
                          />
                        </g>
                      </svg>
                    </span>
                  </div>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="status-page-container"
    >
      <h1
        class="page-title"
        data-testid="title"
      >
        Status Page
      </h1>
      <div
        class="Datatable Datatable--density-default"
      >
        <table
          aria-rowcount="1"
          class="Datatable__table Datatable__table--col-fixed"
        >
          <colgroup>
            <col
              style="width: 33.333333333333336%;"
            />
            <col
              style="width: 50%;"
            />
            <col
              style="width: 16.666666666666668%;"
            />
          </colgroup>
          <thead
            class="Datatable__header"
          >
            <tr
              role="row"
            >
              <th
                aria-sort="none"
                class="header__cell Datatable__cell"
                role="columnheader"
                scope="col"
              >
                <div
                  class="header__name"
                >
                  Service Name
                </div>
              </th>
              <th
                aria-sort="none"
                class="header__cell Datatable__cell"
                role="columnheader"
                scope="col"
              >
                <div
                  class="header__name"
                >
                  Endpoint
                </div>
              </th>
              <th
                aria-sort="none"
                class="header__cell Datatable__cell"
                role="columnheader"
                scope="col"
              >
                <div
                  class="header__name"
                >
                  Status
                </div>
              </th>
            </tr>
          </thead>
          <tbody
            class="Datatable__body"
          >
            <tr
              class="Datatable__row"
            >
              <td
                class="row__cell Datatable__cell"
              >
                <a
                  href="https://github.expedia.biz/eg-reo-opex/eg-fci-data-service"
                  target="_blank"
                >
                  eg-fci-data-service
                </a>
                <img
                  alt="github"
                  height="24"
                  title="github"
                  width="24"
                />
              </td>
              <td
                class="row__cell Datatable__cell"
              >
                /v1/checkout-failures/sites
              </td>
              <td
                class="row__cell Datatable__cell"
              >
                <div
                  data-testid="successIcon"
                >
                  <span
                    class="SVGIcon success-icon"
                  >
                    <svg
                      data-id="SVG_SUCCESS__24"
                      focusable="false"
                      height="24"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <g
                        fill="none"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                      >
                        <path
                          d="M2.8 12a9.3 9.3 0 0 1 12.7-8.5 9.3 9.3 0 0 1 5 12 9.3 9.3 0 0 1-12 5A9.3 9.3 0 0 1 2.8 12z"
                        />
                        <path
                          d="M13.2 12.5l-2.9 2.8-2.8-2.8M16.5 9.1l-3.3 3.4"
                        />
                      </g>
                    </svg>
                  </span>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
